### PR TITLE
Debug logger review

### PR DESCRIPTION
**Pull request by ffrostflame**

### CODE REVIEW

File: src/lib/logger.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+	log(severity: severityLevels, ...toPrint: any[]) {
+		let stringToPrint = "";
+		toPrint.forEach((value) => {
+			stringToPrint = `${stringToPrint}${tostring(value)} `;
+		});
+
+		output.unshift({
+			level: severity,
+			text: `[${this.name}]: ${stringToPrint}`,
+		});
```
Comment (RigidStudios) [inline src/lib/logger.ts:35]: Please don't do this, very bad performance

Comment (ffrostflame) [reply src/lib/logger.ts:35]: I'm confused. Are you talking about the concat, or the unshift?

String interpolation is not performance heavy, and .unshift() is a single table.insert call, which isn't bad performance.

Comment (RigidStudios) [reply src/lib/logger.ts:35]: It's the .unshift, just make it append to end because that's how we're going to be displaying anyway.

**Comment by RigidStudios** [comment]
Change the `ESeverityLevels` name then I'll clear and commit this.

**Comment by RigidStudios** [comment]
@ffrostflame ^

**Review by RigidStudios — APPROVED** [review]
Very cool!

**Pull request merged**